### PR TITLE
add example to new branch

### PR DESCRIPTION
**Pull request by JessicaBertolasi**
Add an example for create prophesee and imu streams for integrate information from an external imu to silky cam from prophesee

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,371 @@
+# CMAKE generated file: DO NOT EDIT!
```
Comment (Jegp) [inline Makefile:1]: I don't think this file should be included. Could you add it to your .gitignore?

File: compile_commands.json [hunk 2]
```diff
@@ -0,0 +1,282 @@
+[
```
Comment (Jegp) [inline compile_commands.json:1]: This file also looks suspicious. Are you certain this should be included?

File: install_manifest.txt [hunk 3]
```diff
@@ -0,0 +1,40 @@
+/usr/include/flatbuffers/minireflect.h
```
Comment (Jegp) [inline install_manifest.txt:1]: This probably shouldn't be added either.

File: src/Makefile [hunk 4]
```diff
@@ -0,0 +1,264 @@
+# CMAKE generated file: DO NOT EDIT!
```
Comment (Jegp) [inline src/Makefile:1]: ... And same for this file.

File: src/input/Makefile [hunk 5]
```diff
@@ -0,0 +1,285 @@
+# CMAKE generated file: DO NOT EDIT!
```
Comment (Jegp) [inline src/input/Makefile:1]: And this :-)

**Review by Jegp — CHANGES_REQUESTED** [review]
First of, the only fifle that should probably be changes is the `example/prophesee_imu.py` file. Everything else should be excluded. Perhaps by adding them to the .gitignore file?\

I do appreciate you wanting to add examples, but I'm not really sure what the present example has to do with aestream. In fact, your code isn't using aestream at all. How about we wait until we've added the code into aestream and **then** write a killer example?

What I could imagine is a way to include some of your code _into_ aestream. But likely in C++ for performance. And there must be a neat way to write this, so we get the function signatures I was referring to in the issue discussion we had.

**Comment by JessicaBertolasi** [comment]
yes I only created the new example, the other files were ready for pull when I added the new file, so I didn't touch em!
Well okay, as I told this was a rude example of my idea for the two streams, but I didn't know how  to integrate with aestream (as I commented in my code) but yes, your suggestion is right. I prefer working with python than c++, that's why I don't know/I'm not sure how to integrate my code with aestream one

**Comment by Jegp** [comment]
I sympathise with your Python preference. :-)

How about this: if you can provide me with a code snippet for extracting a stream of IMU events (either as a comment here or in the PR), then I can try to build a C++ skeleton around it we can merge into aestream together - and interface it from Python. I understand that means working with C++, but since aestream _needs_ to work with C++ to be efficient, that's the minimum requirement to integrate it into our pipeline.

**Comment by JessicaBertolasi** [comment]
this is what I've done for save imu data on txt file for lm9ds1 imu

```
import sys, os
import time
import datetime

import numpy as accTot
import numpy as gyrTot
import numpy as np

sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))
from lsm9ds1_rjg import Driver, I2CTransport, SPITransport

# data write on file
header = 'timestamp'+' '+'Acceleration(x,y,z)'+'  '+'Gyroscope(x,y,z)'+'  '+'Temp'
f = open("IMUflow.txt",'w+') 
f.write(header + "\n")

class SimpleExample:

    def __init__(self):
        self.driver = self._create_i2c_driver()
        self.driver.configure()

    @staticmethod
    def _create_i2c_driver() -> Driver:
        return Driver(
            I2CTransport(1, I2CTransport.I2C_AG_ADDRESS),
            I2CTransport(1, I2CTransport.I2C_MAG_ADDRESS))

    @staticmethod
    def _create_spi_driver() -> Driver:
        return Driver(
            SPITransport(0, False),
            SPITransport(1, True))

    def main(self):

        print("\nIMU data reading ...\n")

        recordArray = np.array([0, 0, 0])

        try:

            while True: 
                ag_data_ready = self.driver.read_ag_status().accelerometer_data_available
                if ag_data_ready:
                    temp, acc, gyro = self.driver.read_ag_data() # get imu flow data
                    ct = datetime.datetime.now() # get timestamp
                    ts = ct.timestamp()
                    recordArray = np.array(np.c_[recordArray, gyro]) 
                    np.savetxt(f,np.array([ts]+acc+gyro+[temp]) [None,:],fmt='%f %i %i %i %i %i %i %i', newline='\n')

                else:
                    time.sleep(0.00001)

        finally:
            self.driver.close()
            recordArray = recordArray.T
            
            f.close()

   
if __name__ == '__main__':
    SimpleExample().main() 
```

**Comment by Jegp** [comment]
Right. I think there's a bit of way to go from here (Python with I2C) to a C++ implementation. It's actually quite straight-forward. You would have to implement a C++ struct that captures the necessary data (a bit like the [PolarityEvent struct](https://github.com/norse/aestream/blob/main/src/aedat.hpp#L18):
```C++
struct IMUEvent { ... };
```
And a function that looks a bit like [the one that generates events from files](https://github.com/norse/aestream/blob/main/src/input/file.cpp#L11):
```C++
Generator<IMUEvent> generate_imu_events(...);
```

With that, we can start figuring out how to make it work with parallel event streams, etc. I don't have the time to start the implementation, but I am happy to guide you if you're still interested in proceeding.

**Comment by Jegp** [comment]
Closing due to inactivity. Please don't hesitate to reopen :-)

**Pull request closed without merging**